src/Entities/Pickup.cpp

```diff
@@ -43,7 +43,7 @@ class cPickupCombiningCallback
 		cItem & Item = OtherPickup.GetItem();
 		if ((Distance < 1.2) && Item.IsEqual(m_Pickup->GetItem()) && OtherPickup.CanCombine())
 		{
-			short CombineCount = Item.m_ItemCount;
+			char CombineCount = Item.m_ItemCount;
 			if ((CombineCount + m_Pickup->GetItem().m_ItemCount) > Item.GetMaxStackSize())
```

Comment:
could we have CombineCount = 64, m_ItemCount = 64, and their sum = 128 overflowing signed `char`?

Reply:
8 Bit = 256
7 Bit = 128

so yeah 64+64 will result in 0. I'm gonna think of something else.

Reply:
we could change the storage of cItem::m_ItemCount to unsigned, might be a bit a work but we were planning on it anyway

Reply:
Unsigned brings it's own pitfalls. We need the compilers to warn on all the arithmetic footguns, then some more sensible semantics might come about.

Reply:
I think I'm going to take this change out of this PR.

And then do the unsigned char change in another and fix this there.

When using unsigned char this should suffice. 8 bits may contain a value of 255 which is enough for this, isn't it?

Reply:
The additions will work, and an item count will ever start off negative. But, we need to verify in places when a subtraction `minuend - subtrahend` happens that `minuend >= subtrahend` otherwise with unsigned the result will wrap instead of going negative.

Reply:
I just realised, the code as written is correct, since the operands of the addition are both `short` to get promoted to `int`, similar with the comparison.